control-plane/agents/src/bin/core/controller/resources/operations.rs

```diff
@@ -149,3 +149,34 @@ pub(crate) trait ResourceShutdownOperations {
         request: &Self::RemoveShutdownTargets,
     ) -> Result<(), SvcError>;
 }
+
+/// Resource Snapshot Operations.
+/// This can be used in combination with `ResourceLifecycle` where the parent resource can use
+/// the `ResourceLifecycle` to manage the snapshot resource.
+#[async_trait::async_trait]
+pub(crate) trait ResourceSnapshotting {
+    type Create: Sync + Send;
+    type CreateOutput: Sync + Send + Sized;
+    type Destroy: Sync + Send;
+    type List: Sync + Send;
+    type ListOutput: Sync + Send + Sized;
+
+    /// Create a snapshot for the `Self` resource.
+    async fn create_snap(
```

Comment:
Do we need the `_snap` suffix ?

Reply:
IMHO keeping the suffix makes it easier to grep for the function if we have to. Otherwise we'd have to search through hundreds of create() functions.

Reply:
Not having the suffix can also make it more difficult to call if you have 2 traits with the same create name implemented for the same type, which happens here.

Reply:
Right. Thanks